### DISCUSSION TITLE
Can I call one Gatsby resolver from another?

### DISCUSSION THREAD

**Discussion by LandonSchropp** [gatsbyjs/gatsby#35008] [Help]
I'm attempting to write a custom Gatsby resolver that combines two different sources: [gatsby-source-notion-api](https://github.com/orlowdev/gatsby-source-notion-api) and [gatsby-plugin-mdx](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/).

Here's what I have so far:

``` javascript
const { createContentDigest } = require("gatsby-core-utils");

function transformNodesToArticle(notionNode, mdxNode) {

  // Build the data for the node.
  let data = {
    id: `${ notionNode.id } >>> Article`,
    title: notionNode.title,
    slug: notionNode.properties["Slug"].value,
    date: notionNode.properties["Date"].value.start,
    description: notionNode.properties["Description"].value,
    url: notionNode.properties["URL"]?.value,
    publisher: notionNode.properties["Publisher"]?.value?.name,
    published: notionNode.properties["Published"].value,
    content: mdxNode.body
  };

  // Create the Gatsby Node structure using the data.
  return {
    parent: null,
    children: [],
    internal: {
      type: "Article",
      contentDigest: createContentDigest(data)
    },
    ...data
  };
}

exports.createSchemaCustomization = ({ actions: { createTypes } }) => {
  createTypes(`
    type Article implements Node {
      id: ID!
      parent: Node!
      children: [Node!]!
      internal: Internal!

      title: String!
      slug: String!
      date: Date!
      description: String!
      url: String!
      publisher: String!
      published: Boolean!
      content: String!
    }
  `);
};

exports.createResolvers = ({ createResolvers }) => {
  createResolvers({
    Query: {
      article: {
        type: "Article",
        resolve: async (source, args, context) => {
          const notionNode = await context.nodeModel.findOne({
            query: {
              filter: {
                properties: {
                  Slug: { value: { eq: args.slug.eq } },
                  Type: { value: { string: { eq: "Article" } } }
                }
              }
            },
            type: "Notion"
          });

          const mdxNode = await context.nodeModel.findOne({
            query: {
              filter: {
                frontmatter: {
                  Slug: {
                    eq: args.slug.eq
                  }
                }
              }
            },
            type: "Mdx"
          });

          return transformNodesToArticle(notionNode, mdxNode);
        }
      }
    }
  });
};
```

This *almost* works. Both queried nodes are returned, and the `Article` node is built correctly. However, `mdxNode.body` is undefined. I think this might be because gatsby-plugin-mdx [uses a resolver for this field](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-mdx/gatsby/create-schema-customization.js#L155-L158).

Is it possible to get the result from gatsby-plugin-mdx's `body` resolver from within my own custom resolver?

**Comment by LekoArts** [comment]
Hi!

This will be a short explanation as writing all the details down would take a while. Feel free to ask any follow-up questions though!

I'd probably do it like this:

- Create a generic interface called `Article`
- Have a `MdxPost` and `NotionPost` type that implements both `Article` and `Node`
- Use a "passthrough" on the `MdxPost` to have the data like `body` on the generic `Article` interface

You can see an example of this here: https://github.com/LekoArts/gatsby-themes/blob/main/themes/gatsby-theme-minimal-blog-core/gatsby-node.js
Then, when e.g. someone wants to use `YAML` they can implement the `YAML` type into the interface like this: https://codesandbox.io/s/minimal-blog-other-data-source-3bm8z?file=/gatsby-node.js
